libs/agno/agno/os/auth.py

```diff
@@ -55,3 +58,175 @@ def validate_websocket_token(token: str, settings: AgnoAPISettings) -> bool:
 
     # Verify the token matches the configured security key
     return token == settings.os_security_key
+
+
+def require_scopes(required_scopes: List[str]):
+    """
+    Create a FastAPI dependency that checks for required scopes.
+
+    This is a helper function for future use when you want to add
+    scope requirements directly to specific endpoints using FastAPI's
+    Depends() mechanism.
+
+    Usage:
+        @router.get("/endpoint", dependencies=[Depends(require_scopes(["scope:read"]))])
+        async def my_endpoint():
+            ...
+
+    Args:
+        required_scopes: List of scopes required to access the endpoint
+
+    Returns:
+        A dependency function that validates scopes
+    """
+
+    def scope_checker(request: Request) -> bool:
+        # Check if user is authenticated
+        if not getattr(request.state, "authenticated", False):
+            raise HTTPException(
+                status_code=401,
+                detail="Authentication required",
+            )
```

Comment:
Should this happen here? we should always be running this when authorization=True in the OS, not only when checking for scopes

Reply:
This whole function is redundant and not used, removing